src/App/App.tsx

```diff
@@ -1,184 +1,106 @@
-import React, { Component, ReactElement } from 'react';
+import React, { useState, useEffect } from 'react';
+import styled from '@emotion/styled';
 import isNil from 'lodash/isNil';
+import { Router } from 'react-router-dom';
 
 import storage from '../utils/storage';
-import { makeLogin, isTokenExpire } from '../utils/login';
-import Loading from '../components/Loading';
-import LoginModal from '../components/Login';
-import Header from '../components/Header';
-import { Container, Content } from '../components/Layout';
+import { isTokenExpire } from '../utils/login';
 import API from '../utils/api';
+import Header from '../components/Header';
 import Footer from '../components/Footer';
+import Box from '../muiComponents/Box';
+import Loading from '../components/Loading';
+import StyleBaseline from '../design-tokens/StyleBaseline';
+import { breakpoints } from '../utils/styles/media';
+
+import AppContextProvider from './AppContextProvider';
+import AppRoute, { history } from './AppRoute';
+
+const StyledBoxContent = styled(Box)({
+  padding: 15,
+  [`@media screen and (min-width: ${breakpoints.container}px)`]: {
+    maxWidth: breakpoints.container,
+    width: '100%',
+    marginLeft: 'auto',
+    marginRight: 'auto',
+  },
+});
+
+/* eslint-disable react/jsx-max-depth */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable react/jsx-no-bind */
+/* eslint-disable react-hooks/exhaustive-deps */
+const App: React.FC = () => {
+  const [user, setUser] = useState();
+  const [packages, setPackages] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-import AppRoute from './AppRoute';
-import { AppProps, AppContextProvider } from './AppContext';
-
-export default class App extends Component<{}, AppProps> {
-  public state: AppProps = {
-    logoUrl: window.VERDACCIO_LOGO,
-    user: {},
-    scope: window.VERDACCIO_SCOPE || '',
-    showLoginModal: false,
-    isUserLoggedIn: false,
-    packages: [],
-    isLoading: true,
+  /**
+   * Logouts user
+   * Required by: <Header />
+   */
+  const logout = () => {
```

Comment:
Some recommendations with the functions inside this hook:

- `useCallback`: for `logout`, `loadOnHandler`

It improve performance.

Reply:
True. Thank you!! I will update it then :-)